JSON/JSON (Basic).sublime-syntax

```diff
@@ -0,0 +1,388 @@
+%YAML 1.2
+---
+#   YAML Documentation:
+#       https://yaml.org/spec/1.2/spec.html
+#   Sublime Text Documentation:
+#       https://www.sublimetext.com/docs/syntax.html#ver-dev
+#       https://www.sublimetext.com/docs/syntax.html#testing:ver-dev
+#       https://www.sublimetext.com/docs/scope_naming.html
+#   This file is being maintained at:
+#       https://github.com/sublimehq/Packages/blob/master/JSON/JSON%20(Basic).sublime-syntax
+#
+name: JSON (Basic)
+scope: source.json.basic
+version: 2
+hidden: true
+
+
+variables:
+  exponent: (?:[eE][-+]?\d+)
+  pos_integer_decimal: (?:0|[1-9]\d*)
+
+
+contexts:
+
+  prototype:
+    - include: comments
+
+  main:
+    - match: '(?=\S)'
+      push: top-level-expect-object
+
+####[ Top level ]#######################################################################################################
+
+  # Try these below one-by-one.
+
+  top-level-expect-object:
+    - include: top-level-objects
+    - match: ''
+      push: top-level-expect-array
+
+  top-level-expect-array:
+    - include: arrays
+    - match: ''
+      push: top-level-expect-constant
+
+  top-level-expect-constant:
+    - include: constants
+    - match: ''
+      push: top-level-expect-number
+
+  top-level-expect-number:
+    - include: numbers
+    - match: ''
+      push: top-level-expect-string
+
+  top-level-expect-string:
+    - include: strings
+    - match: ''
+      push: top-level-expect-invalid-remainder
```

Comment:
This causes several contexts to be pushed on stack, once "string" is used as top-level item for instance.

![grafik](https://user-images.githubusercontent.com/16542113/144706221-09d960db-81dd-4e96-8d8c-6672d63f52cd.png)

I don't get the use of this construct.

Reply:
In JSON only one of these (constructs or values) can be at the top level:

- object
- array
- constant
- number
- string

Anything else following that is invalid. Having:

```json
"one"
"two"
```

I would mark `"one"` as valid JSON, but anything afterwards as `invalid`.

Reply:
So this was the approach of trying them one-by-one.

Reply:
Maybe that context stack could be avoided if it would `set` instead of `push`? Or just include those contexts if they already `set` inside them?

Reply:
How about `"string""string"`? Second one should be invalid shouldn't it?

Reply:
So do I `push` from `main` and then use `set` to go through the rest?

Reply:
I am experimenting with creating popping variants of array/string/... to match only one at top-level. This strategy might be useful in values as well.

Reply:
Even `"one", "two"` at the top level would be invalid from the comma until the end.

Reply:
I'll give it a try.

Reply:
Thanks for the input.

Reply:
Before I try, I assume this will mark everything as `invalid.illegal.unexpected-code-after-first-structure-or-value.json` in any case? So maybe not.

Reply:
Added my 5ct about it to https://github.com/jrappen/sublime-json/pull/8

Reply:
Thanks, I'll take a look at this tonight.

Reply:
Part of next push thanks to deathaxe.